tensorflow/python/saved_model/save_test.py

```diff
@@ -667,23 +666,5 @@ def call(self, x, y):
     return {"loss": loss}
 
 
-class MemoryTests(test.TestCase):
```

Comment:
Curious about the removal of this test suite

Reply:
Yeah that was my suggestion since the test was failing (I wrote the test). tf.cond creates reference cycles, so if we use tf.cond when saving then the test is broken.

The satisfying fix is to make tf.cond not do that. It's not intractable, we've done it for tf.function. But I don't think we need to block this change on that fix. 

I think the reference cycles should just be symbolic tensors, so not holding on to device memory waiting for Python's GC. Although it's a slight risk.

Reply:
Thank you. Now it makes sense.